File: BitWiseManipulation/Java/CountFlipBit.java

import java.util.Scanner;

public class CountFlipBit {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter the two numbers: ");
        int a = sc.nextInt();
        int b = sc.nextInt();
        int count = 0;
        for (int c = a ^ b; c != 0; c = c & (c - 1)) {
            count++;
        }
        System.out.println(count);
    }
}

File: BitWiseManipulation/C++/CountFlipBit.cpp
#include <iostream>
using namespace std;

int countBitsToFlip(int A, int B) {
    int xorResult = A ^ B;
    int count = 0;
    while (xorResult > 0) {
        count += (xorResult & 1);
        xorResult >>= 1;
    }
    return count;
}

int main() {
    int A, B;
    cout << "Enter two numbers A and B: ";
    cin >> A >> B;
    cout << "Number of bits to flip: " << countBitsToFlip(A, B) << endl;
    return 0;
}
